arkouda/timeclass.py

```diff
@@ -144,6 +139,131 @@ def __init__(self, array, unit: str = _BASE_UNIT):  # type: ignore
+            days = cast((seconds // 86400), int64) + 1
+            self._year = cast((days // 365.25 + 1970), int64)
+
+            self._hour = cast((self.values / _get_factor("h")), int64)
+            self._hour %= 24
+            self._minute = cast((self.values / _get_factor("m")), int64)
+            self._minute %= 60
+            self._second = cast((self.values / _get_factor("s")), int64)
+            self._second %= 60
+
+            # Month, day
+            self._day = self.values // _get_factor("d")
+            self._day += 1401  # move starting point to 1996 - 03 - 01
+            self._day %= 1461  # max 3 years 365 day (not 4 years!) remaining
+            d = self._day[:]
+            self._day[d >= 730] -= 730
+
+            d = self._day[:]
+            self._day[d >= 365] -= 365
+
+            d = self._day[:]
+            self._month = d[:] - d[:] + 3
+            self._day[d >= 306] -= 306
+            self._month[d >= 306] = 1
+
+            d = self._day[:]
+            self._day[d >= 153] -= 153
+            self._month[d >= 153] = 8
+
+            d = self._day[:]
+            self._day[d >= 122] -= 122
+            self._month[d >= 122] += 4
+            d = self._day[:]
+            self._day[d >= 61] -= 61
+            self._month[d >= 61] += 2
+
+            d = self._day[:] + 1
+            self._day[d >= 32] -= 31
+            self._month[d >= 31] += 1
+            self._day += 1
+            self._day[d >= 30] += 1
+            self._day[self._day >= 32] = 1  # get the day of the month
+            self._month[self._month >= 13] = 1
+
+            # Day of year
+            month_days = [
+                [0, 0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334],
+                [0, 0, 31, 60, 91, 121, 152, 182, 213, 244, 274, 305, 335],
+            ]
+
+            year = self._year.to_ndarray()
+            lyear = self._year[:].to_ndarray()
+            lyear = [np.logical_and(year % 4 == 0, year % 100 != 0)]
+            lyear = np.array([np.logical_or(lyear, year % 400 == 0)]).astype(int)
+            days = self._day[:].to_ndarray()
+            self._date = []
+            for i in range(days.size):
+                days[i] += month_days[lyear[0][0][i]][self._month[i]]
+                self._date.append(f"{self._year[i]}-{self._month[i]:02d}-{self._day[i]:02d}")
+            self._dayofyear = ak.array(days)
+
+            # Day of week
+            self._dayofweek = cast(((self.values / _get_factor("d")) + 3), int64)
+            self._dayofweek %= 7
+
+    # Datetime values
+    @property
+    def date(self):
+        return self._date
+
+    @property
+    def year(self):
+        return self._year
+
+    @property
+    def month(self):
+        return self._month
+
+    @property
+    def day(self):
+        return self._day
+
+    @property
+    def week(self):
+        return self._week
+
+    @property
+    def dayofyear(self):
+        return self._dayofyear
```

Comment:
The return here is shifted
```python
start_time = "2022-07-18 12:59:00"
dt = ak.date_range(start=start_time, periods=5, freq="d")
x = dt.dayofyear
```
results in 
```
[198 199 200 201 202]
```

pandas equivalent results in
```
Int64Index([199, 200, 201, 202, 203], dtype='int64')
```

Reply:
Fixed algorithm